### ISSUE TITLE
Don't do configuration/setup at the top level

### ISSUE THREAD

**Issue by noctuid** [seagle0128/doom-modeline#142]
**Describe**
Doom modeline does stateful configuration/setup at the top level.

**Steps and Expected**
Run `(require 'doom-modeline)`, and `(eldoc-in-minibuffer-mode 1)` is enabled. Similarly, various functions are added to hooks and advice is created. For example:
```elisp
(add-hook 'window-configuration-change-hook #'doom-modeline-set-selected-window)
(advice-add #'handle-switch-frame :after #'doom-modeline-set-selected-window)
(advice-add #'select-window :after #'doom-modeline-set-selected-window)
(advice-add #'make-frame :after #'doom-modeline-set-selected-window)
(advice-add #'delete-frame :after #'doom-modeline-set-selected-window)
(with-no-warnings
  (cond ((not (boundp 'after-focus-change-function))
         (add-hook 'focus-in-hook  #'doom-modeline-set-selected-window)
         (add-hook 'focus-out-hook #'doom-modeline-unset-selected-window))
        ((defun doom-modeline-refresh-frame ()
           (setq doom-modeline-current-window nil)
           (cl-loop for frame in (frame-list)
                    if (eq (frame-focus-state frame) t)
                    return (setq doom-modeline-current-window (frame-selected-window frame)))
           (force-mode-line-update))
         (add-function :after after-focus-change-function #'doom-modeline-refresh-frame))))
```

This setup should only be done *after* the user has enabled doom modeline, not when loading the package. All hooks and advice should be added when enabling and removed when disabling. `eldoc-in-minibuffer-mode` should be enabled when enabling doom modeline and should be disabled when disabling doom modeline only if it was previously disabled.

**Comment by seagle0128** [comment]
Thanks for you reporting! I think I understand what you mean completely. I will try to refactor the codes, but I don't think it's deserved to change all of them since they are minor impacts. Actually before only `doom-modeline-init` is provided, and no way to remove it dynamically. `doom-modeline-mode` is for better troubleshooting and is not designed and recommended to change frequently.

**Comment by seagle0128** [comment]
Closed by 9bc96d908f0606e72ebcc8b43303fb615f623a3c
